var/spack/repos/builtin/packages/opencv/package.py

```diff
@@ -107,7 +118,6 @@ class Opencv(CMakePackage):
     depends_on('qt', when='+qt')
     depends_on('java', when='+java')
     depends_on('py-numpy', when='+python', type=('build', 'run'))
-    depends_on('protobuf@3.1.0', when='@3.3.0: +dnn')
```

Comment:
Is there a way to tell opencv not to build its own protobuf? Allowing packages to build their dependencies is entirely against the Spack model and often causes dependency incompatibility problems when we compose dependencies into larger packages.

Reply:
You're right. I found a corresponding flag and added it to the package. Therefore, I reinserted the dependency again (considering that the current opencv version requires another protobuf version)